### ISSUE TITLE
Content-DPR makes for an imperfect placeholder

### ISSUE THREAD

**Issue by eeeps** [eeeps/content-dpr-explainer#4]
See my description of the problem and a bunch of ideas about potential solutions here: https://github.com/httpwg/http-extensions/issues/1102

TL;DR:

- Rounding errors w/ Content-DPR will often cause small layout jumps, in transitions between a placeholder and a full-resolution image
- More precision would allow us to solve this in one dimension only. More precision could be achieved by splitting the header value into integer and fractional parts (`Content-DPR: int=1, frac=234567`), or into a rational (`Content-DPR: a=1234567, b=1000000`).
- Specifying exact target intrinsic dimensions would allow us to solve it in both dimensions (for instance `Content-DPR: w=871, h=576` or even `Content-DPR: 1.234567;w=871;h=576;`).

For this particular use case, exact dimensions would be best; failing that, more precision would be good; failing that, the current limitations... might be acceptable? Some things I'm noodling on:

- Not sure if intrinsic dimensions are confusingly outside of the scope of the `Content-DPR` name, or still aligned with the other use cases, such that everything still fits under the same umbrella.
- Not sure if this use case is important enough to merit it's own syntax or header.
- Not sure if the existing behavior, with small layout jumps, is "good enough."
- Pretty sure it would be harder to integrate exact intrinsic dimensions into HTML (this is not a dealbreaker, just work)

**Comment by noamr** [comment]
> * Pretty sure it would be harder to integrate exact intrinsic dimensions into HTML (this is not a dealbreaker, just work)
I don't think so. At least in webkit exact intrinsic dimesnsions would make more sense than (or just as much sense as) dividing by DPR and then dividing again by 1 / Content-DPR.
